pkg/trace/sampler/coresampler.go

```diff
@@ -303,13 +268,6 @@ func (s *Sampler) size() int64 {
 	return int64(len(s.seen))
 }
 
-func (s *Sampler) report() {
-	s.metrics.report()
-	_ = s.statsd.Gauge(metricSamplerSize, float64(s.size()), s.tags, 1)
-}
-
-// Stop stops the main Run loop
-func (s *Sampler) Stop() {
-	close(s.exit)
-	<-s.stopped
+func (s *Sampler) report(statsd statsd.ClientInterface, name Name) {
+	_ = statsd.Gauge(MetricSamplerSize, float64(s.size()), []string{"sampler:" + name.String()}, 1)
```

Comment:
Not sure how often this is called, but we're incurring in new allocations here with:
```
[]string{"sampler:" + name.String()}
```
which could have some performance impact. If this is not in a hot-path it's probably OK, otherwise it might be worth precomputing the tags and passing them to `report()`.

Reply:
Looks like every 10s, so should be OK.